include/xtensor/xinfo.hpp

```diff
@@ -62,14 +62,6 @@ namespace xt
     {
         std::string s;
         using shape_type = typename T::shape_type;
-        if (detail::is_array<shape_type>::value)
-        {
-            s += "Type: xtensor, fixed dimension " + std::to_string(t.dimension());
-        }
-        else
```

Comment:
Is there another interface we could rely on? E.g. maybe we could check wether the shape_type contains a function like `resize`? 
This could use the same mechanism as `has_raw_data` or some other magic.

Reply:
Or we could display the actual typename for the expression.

Reply:
I would suggest we do it in a different PR.

Reply:
yes, for sure in a different PR. 

Actually, displaying `T` also prints some superflous characters on recent GCC versions, so we might want to adjust that, too (in xinfo).